machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +64,26 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
-	// Preemptible indicates if created instance is preemptible
+	// GPUs is a list of GPUs to be attached to the VM.
+	// +optional
+	GPUs []GCPGPUConfig `json:"gpus,omitempty"`
+	// Preemptible indicates if created instance is preemptible.
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// This is required to be set to "Terminate" if you want to provision machine with attached GPUs.
+	// Otherwise, allowed values are "Migrate" and "Terminate".
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is "Migrate".
+	// +kubebuilder:validation:Enum=Migrate;Terminate;
+	// +optional
+	OnHostMaintenance GCPHostMaintenanceType `json:"onHostMaintenance,omitempty"`
+	// AutomaticRestart determines the behavior when an instance crashes or the underlying infrastructure provider stops the instance as part of a maintenance event (default "Always").
+	// Cannot be "Always" with preemptible instances.
+	// Otherwise, allowed values are "Always" and "Never".
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is "Always".
+	// +kubebuilder:validation:Enum=Always;Never;
+	// +optional
+	AutomaticRestart corev1.RestartPolicy `json:"automaticRestart,omitempty"`
```

Comment:
I think we should change the field to be called `RestartPolicy` and define our own enum values. Using the `corev1.RestartPolicy` enum does work, except there are 3 possible values there, and currently, we only support 2 possible values

Reply:
I considered this, however i assumed we do not want to rename the field, since we do not have docs concerning the `providerSpec` and therefore it might be confusing for customer if the field does not match the GCP docs. 

I can make the change and mention it in the comment that this `RestartPolicy` field is reffering to `AutomaticRestart` in google docs.